src/Typography/Typography.component.tsx

```diff
@@ -88,9 +86,14 @@ export interface StyledTypographyProps extends TypographyProps {
     $color?: 'inherit' | 'initial' | string;
 }
 
-const StyledTypography = (tag: TypographyTags) => styled(tag)<
-    StyledTypographyProps
->`
+export const Typography = styled('span').attrs<TypographyProps>(
```

Comment:
Converting Typography to a `styled` component (rather than a component that returns a styled component) allows styled-components to squash the styles of components that styled-extend Typography (i.e.  styled(Typography)).

The issue with the missing `anchor-typography` classes was with components that both extend Typography (e.g. Hero.Title, Alert, List.Item) and use the `as` prop. The explanation is complicated (https://github.com/styled-components/styled-components/issues/1981#issuecomment-419287003) and is also the expected behavior for styled-components. This change fixes that issue.

Reply:
I read through this and it's a known issue and while I don't like that we lose the ability to access typography by it's class under certain conditions, I don't think this is a gate for this PR.

Reply:
Also, thanks for finding this and spending the time to explain it.